### PR TITLE
Remove debug options from AHC-WS

### PR DESCRIPTION
**Pull request by wsargent**
## Purpose

This removes the calls in `AHCWSModule` to SSLConfig's `DebugConfiguration` and `SystemConfiguration` which does evil monkeypatching which does not work in Java 9+.

SSLConfig will have a trace debug wrapper that works with plain Java.

## Background Context

* https://github.com/lightbend/ssl-config/issues/142

### CODE REVIEW

**Comment by dwijnand** [comment]
I'd rather only remove these if they have a replacement.

* What happens when a user has `wsClientConfig.ssl.debug.enabled` set?
* How is the system configurated without `new SystemConfiguration(loggerFactory).configure(wsClientConfig.ssl)`?

**Comment by wsargent** [comment]
> What happens when a user has wsClientConfig.ssl.debug.enabled set?

The replacement is in https://github.com/lightbend/ssl-config/pull/143 but that's in a distinct project, so hard to move.

> How is the system configurated without new SystemConfiguration(loggerFactory).configure(wsClientConfig.ssl)?

That should be handled through system properties -- in practice that means we go through `fork in run := true` / `javaOptions="-Djavax.net.debug=all"` and go back to https://docs.oracle.com/javase/7/docs/technotes/guides/security/jsse/ReadDebug.html.  The system configuration inside JVM has the same race condition issues in that if the class is loaded before the system properties are set, then it's too late to reset them.

**Comment by wsargent** [comment]
Should be addressed by https://github.com/lightbend/ssl-config/pull/143, but will need a merge into play-ws.

**Comment by dwijnand** [comment]
@marcospereira what do you think of this change?  Should we rebase it and merge it for Play 2.8?

**Comment by marcospereira** [comment]
Hey @wsargent I have a local rebase with a small change in the migration guide so that we can land this change, but I cannot push to your branch. Could you give me push rights to your branch? Of course, if you prefer, I can do the migration change later in another PR.

Best.

**Comment by wsargent** [comment]
@marcospereira done, you should have an invite link in email

**Comment by marcospereira** [comment]
Thanks, @wsargent. Rebased and forced push there.

File: documentation/manual/releases/release28/migration28/Migration28.md [hunk 1]
```diff
@@ -90,6 +90,14 @@ Many changes have been made to Play's internal APIs. These APIs are used interna
 
 ## Configuration changes
 
+This section lists changes and deprecations in configurations.
+
+### Debugging SSL Connections
+
+Until Play 2.7, both Play and Play-WS were using a version of [ssl-config](https://lightbend.github.io/ssl-config/) which had a debug system that relied on undocumented modification of internal JSSE debug settings. These are usually set using `javax.net.debug` and `java.security.debug` system properties on startup.
+
+This debug system has been removed, the debug flags that do not have a direct correlation in the new system are deprecated, and the new configuration is documented in [ssl-config docs](https://lightbend.github.io/ssl-config/DebuggingSSL.html).
```
Comment (dwijnand) [inline documentation/manual/releases/release28/migration28/Migration28.md]: :+1:

**Pull request merged**